packages/client/src/components/interface/DownloadButton.tsx

```diff
@@ -262,7 +262,10 @@ function DownloadButtonComponent(props: DownloadButtonProps & HOCProps) {
       if (
         (assignment?.userId !== userId ||
           status === DOWNLOAD_STATUS.DOWNLOADED) &&
-        (downloadConfigs.declarationStatus !== 'VALIDATED' ||
+        ((downloadConfigs.declarationStatus &&
+          !['VALIDATED', 'CORRECTION_REQUESTED'].includes(
```

Comment:
Think it would work okay but still we need to improve this expression at some point (not necessarily as part of this PR)  as this has become kind of hard to follow